File: problems/0202-happy-number/solution.java
class Solution {
    public boolean isHappy(int n) {
       while(n!=1 && n!=4)
       {
        int sum=0;
    
     while(n>0)
     {
        int b=n%10;
        int c=b;
        n=n/10;
        int d=c*c;
        sum=sum+d;
        
            
        }
        
            
                n=sum;
            

     }
     return n==1;

    }
}

File: problems/0202-happy-number/solution.py
class Solution:
    def dig(x):
        lst=[]
        while x:
            lst.append((x%10)**2)
            x=x//10
        return lst
    
    def isHappy(self, n: int) -> bool:
        f=False
        lst =[]
        while True:
            n=sum(Solution.dig(n))
            if n in lst and n!= 1:
                return False
            lst.append(n)
            
            
            
            if n==1:
                return True
                
            
            
            
                
        
        
            
            
